### PR TITLE
Table of contents with {% include toc_sidepanel.start %} / {% include toc_sidepanel.end %}

### PR DESCRIPTION
**Pull request by yonojoy**
Despite the drafts page, I needed some time to figure out how to get the ToC right. I think these includes could help to get the html code out of the markdown file and to help newbies to add a ToC.

### CODE REVIEW

**Comment by TWiStErRob** [comment]
I had the same idea to extract toc display. I think you didn't get this right. TOC is autogenerated and content shouldn't be inside the toc, therefore no need for start/end includes. Here's my take on it:
- https://raw.githubusercontent.com/TWiStErRob/twisterrob.github.io/master/_includes/toc.md
- Blog post: [source](https://raw.githubusercontent.com/TWiStErRob/twisterrob.github.io/master/_posts/2011-10-21-effective-at-work.md) and [rendered](http://www.twisterrob.net/blog/2011/10/effective-at-work.html)
- Collection item: [source](https://raw.githubusercontent.com/TWiStErRob/twisterrob.github.io/master/_projects/inventory.md) and [rendered](http://www.twisterrob.net/project/inventory/)

Also you can set the toc title parameter on the include as an optional argument if you modify my solution: based on how `include.panel` works. You'll likely want to have the same title on most TOC's and not repeat the title every time.

**Comment by yonojoy** [comment]
The toc is on a side panel.  
Start and end are necessary to position the panel. However all the content is indexed.
I didn't change the html code, only remove the html from drafts to include.

Inspired by your code I modified title to be a parameter of the include with default "Table of Contents".

I think both includes mine and yours can be useful depending on the situation. Or did I miss something and it is possible with your solution to have the TOC on the sidebar?

File: _includes/toc_sidepanel.start [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+* # My Content
+* {% include toc_sidepanel.end %}
+* # Other content (gets indexed too) 
+*
+* The whole content is indexed, start and end determine where the panel is placed.
+* If you don't specify title "Table of Contents" is used
+{% endcomment %}
+
+<div class="row">
+<div class="medium-4 medium-push-8 columns" markdown="1">
```
Comment (TWiStErRob) [inline _includes/toc_sidepanel.start:18]: I think you could remove that markdown.

**Comment by TWiStErRob** [comment]
Oh, I see now the "# My Content" confused me because it is just a header, but no paragraph. It looked like "here you can enter your TOC manually" for me.

I went with top-toc because of sidebar blindness. Yes, you need to close the div if you include a sidebar. I want to move the TOC generation to my page template somehow so not all of the posts have to include the layout and stuff, but didn't have time and drive to try to tackle it yet.

**Comment by ghost** [comment]
FWIW Hugo mitigated problems with TOC generation by adding it to core. Jekyll should do the same, or include a standard plug-in for TOC generation for use in GH pages if one is not already provided

**Comment by Phlow** [comment]
in my opinion the toc-version of Kramdown is sufficient.

**Pull request closed without merging**